project.md

```diff
@@ -7,62 +7,195 @@ good way to meet their needs.
 
 ### Why a language?
 
-_Why is a DSL appropriate for your user(s)? How does it address the need?_
+Since this is a visual domain, I can see how a language may seem like an
+ineffective tool to use rather than a GUI, but I believe there are advantages
+the language format provides.
+Firstly, I really like components of my presentation to be constrained precisely
+how I want them, so I think a language is more suitable for this.
+Next, this is using web-based output, so those familiar with web development
+will be familiar with this process of programming components and viewing the
+output in a browser.
+Finally, I really want the resulting program to not feel limiting, so if the
+user wants to add features beyond what I am able to incorporate into the
+program, I think they should be able to simply edit the output HTML/CSS/JS with
+their desired behavior.
+If this is a language, it makes that process much more seamless since they will
+already be working with certain variable names and know what attributes
+different components have, so they can easily find what they want to edit.
 
 ### Why you?
 
-_What excites you about this idea? How did you come up with it?_
+I am really interested in seeing if I can find a way to make presentations
+completely different than the rigid slide-based format of every other software.
+I also love web design and trying to make websites aesthetically pleasing, so
+this project is something that I will find both satisfying and useful.
+There wasn't really any particular thing that sparked this idea for me.
+I think just generally, I have been dissatisfied with the structure of
+presentations and have been considering ways to make them more understandable
+and engaging for a while now.
+As a bonus, I am also a bit dissatisfied by how most websites are layed out with
+a bunch of text or graphics layed out linearly (although I know there's more
+variety in websites than there are in presentations), so this could be an
+alternative tool for creating websites as well.
 
 ### Domain
 
-_Describe the project's domain in five words._
+Dynamic web-based presentation design
 
 ### Interface (syntax)
 
-_How might the user interact with the language? What does programming look
-like? Why is this the right way to interact with the problem domain?_
+How I am imagining it is that the user can add _components_ and _behaviors_ to
+their presentation, and then immediately open the result as a webpage.
+The webpage could have a frame selector at the bottom for easy navigation
+between keyframes.
+This resulting webpage would be completely usable as the final presentation.
+As they are developing, they can refresh the webpage to see the updates they
+make, but this would also require the user to render the program each time they
+want to check.
+I want the experience to be similar to web development, except, unfortunately,
+the web developer tools won't work with my language since it is external.
+
```

Comment:
I think you should look into interactive videos because dynamic presentations are similar to those.